app/product/[productId]/page.tsx

```diff
@@ -1,176 +1,123 @@
-'use client'
-
-import { useEffect, useMemo, useState } from 'react';
-import { Sora } from 'next/font/google'
-import Image from 'next/image'
-import dynamic from "next/dynamic";
+import { Sora } from 'next/font/google';
+import Image from 'next/image';
 import axios from 'axios';
 
-import ProductCard from '@/components/ProductCard'
-import { BuyIcon, MinusIcon, PlusIcon } from '@/components/icons';
-import { ProductItem } from '@/utils/product';
-import Spinner from '@/components/Spinner';
-import Color from '@/utils/color';
-const StarRatings = dynamic(() => import("react-star-ratings"), {
-  ssr: false,
-});
-
-
-const sora = Sora({ subsets: ['latin'] })
+import ProductCard from '@/components/ProductCard';
+import ProductStarRatings from '@/components/ProductStarRatings';
+import ChatbotButtonsProduct from '@/components/ChatbotButtonsProduct';
 
-export default function Detail({ params: { productId } }: { params: { productId: string } }) {
-  const [amount, setAmount] = useState<number>(1);
-  const [product, setProduct] = useState<ProductItem>();
-  const [recommendProducts, setRecommendProducts] = useState<ProductItem[]>();
-  const [detailLoading, setDetailLoading] = useState<boolean>(false);
-  const [recommendLoading, setRecommendLoading] = useState<boolean>(false);
-
-  const productDetails = useMemo(() => {
-    if (product?.product_specification) {
-      const specifications = product.product_specification.split("|");
-      return specifications.map((spec) => {
-        const pair = spec.split(":");
-        return {
-          key: pair[0],
-          value: pair[1]
-        }
-      })
-    }
-  }, [product?.product_specification]);
+import { ProductItem } from '@/utils/product';
 
-  const getProduct = async (productId: string) => {
-    setDetailLoading(true);
-    try {
-      const response = await axios.get<ProductItem>(`/api/product/${productId}`);
-      setProduct(response.data);
-    } catch (err) {
+const sora = Sora({ subsets: ['latin'] });
 
-    }
-    setDetailLoading(false);
-  }
+export default async function Page({
+  params,
+}: {
+  params: { productId: string };
+}) {
+  const { productId } = params;
 
-  const getRecommendProducts = async (productId: string) => {
-    setRecommendLoading(true);
-    try {
-      const response = await axios.get<ProductItem[]>(`/api/product/${productId}/recommended`);
-      setRecommendProducts(response.data);
-    } catch (err) {
+  const productResponse = await axios
+    .get<ProductItem>(`/api/product/${productId}`)
+    .catch((err) => {
+      console.log('err', err);
+    });
 
-    }
-    setRecommendLoading(false);
-  }
+  const recommendProductsResponse = await axios
+    .get<ProductItem[]>(`/api/product/${productId}/recommended`)
+    .catch((err) => {
+      console.log('err', err);
+    });
```

Comment:
I think this is the cause of the errors on the products page i don't think axios calls can be made from a server component

Reply:
I tried, we can hit axios calls.
As product page is now server-side, we need the complete api base url to fetch data, so we need to add env for that. I added an .env.example file.